src/components/Search.vue

```diff
@@ -36,12 +84,38 @@ export default {
     searchParams() {
       return {
         q: this.searchText,
-        mode: this.searchMode,
-        types: JSON.stringify(this.searchTypes),
+        mode: this.internalSearchMode,
+        types: JSON.stringify(this.internalSearchTypes),
         // + 1 to determine if total results > maxQuickResults
         limit: this.maxQuickResults + 1,
       };
     },
+    internalSearchMode: {
+      get() {
+        return this.lazySearchMode;
+      },
+      set(val) {
+        this.lazySearchMode = val;
+        this.$emit('update:searchMode', val);
+      },
+    },
+    internalSearchTypes: {
+      get() {
+        return this.lazySearchTypes;
+      },
+      set(val) {
+        this.lazySearchTypes = val;
+        this.$emit('update:lazySearchTypes', val);
```

Comment:
Could we name this event just `update:searchTypes`? It seems that the internal nature of it is more salient than the lazy part.

Reply:
Done.